### PR TITLE
Skip ACL when no user or empty username is found

### PR DESCRIPTION
**Pull request by githubapreinders**
When the list of queues is browsed tibco returns queuenames and a principal; when the principal is empty an exception is given back. At this point the whole queues-summary cannot be displayed. This is the error:

java.util.concurrent.ExecutionException: nl.nn.adapterframework.core.PipeRunException: Pipe [getTibcoQueues] msgId [0ace000c-3a81a2_17a1ea5a2c1_-7d92] exception on showing Tibco queues, url [tcp://DEVP2PLARGEDC1:37222,tcp://DEVP2PLARGEDC2:37222|tcp://DEVP2PLARGEDC1.insim.biz:37243,tcp://DEVP2PLARGEDC2.insim.biz:37243]: (TibjmsAdminInvalidNameException) Error retrieving user, invalid name

After setting logmessages in an overridden GetTibcoQueues.java the empty principal was given back as an empty string (and not as null)

The adapted code works and ignores a queue that has no recognizable principalname. These queues will not be displayed on the summary pages. It seems that on the tibco side of things, an empty principal can be given back as a very rare exception. This has not happened on ACC and PRD environments but would block the GetTibcoQueues service.

Please review and give feedback.

### CODE REVIEW

File: tibco/src/main/java/nl/nn/adapterframework/extensions/tibco/GetTibcoQueues.java [hunk 1]
```diff
@@ -561,6 +561,8 @@ private Map getAclMap(TibjmsAdmin admin, LdapSender ldapSender) throws TibjmsAdm
 					userMap.put(principal, principalDescription);
 				}
 			}
+			else {
+				continue;}
```
Comment (nielsm5) [inline tibco/src/main/java/nl/nn/adapterframework/extensions/tibco/GetTibcoQueues.java:565]: Ik denk dat je beter een check kan doen als `if(StringUtils.isEmpty(principal)) { continue; }` dat leest wat makkelijker.

Comment (githubapreinders) [reply tibco/src/main/java/nl/nn/adapterframework/extensions/tibco/GetTibcoQueues.java:565]: Ik heb er if(StringUtils.isNotEmpty(principal)) { continue; } van gemaakt. Dat scheelt een extra conditie.

File: tibco/src/main/java/nl/nn/adapterframework/extensions/tibco/GetTibcoQueues.java [hunk 2]
```diff
@@ -589,8 +563,7 @@ private String getLdapPrincipalDescription(String principal, LdapSender ldapSend
 				}
 			}
 		} catch (Exception e) {
-			log.debug("Caught exception retrieving description for principal ["
-					+ principal + "]: " + e.getMessage());
+			log.debug("Caught exception retrieving description for principal [" + principal + "]: " + e.getMessage());
```
Comment (gvanbrakel) [inline tibco/src/main/java/nl/nn/adapterframework/extensions/tibco/GetTibcoQueues.java:566]: hier graag `, e` i.p.v. ` + e.getMessage()`

File: tibco/src/main/java/nl/nn/adapterframework/extensions/tibco/GetTibcoQueues.java [hunk 3]
```diff
@@ -601,16 +574,14 @@ private String getResolvedUrl(String url) {
 		try {
 			uri = new URI(url);
 		} catch (URISyntaxException e) {
-			log.debug("Caught URISyntaxException while resolving url [" + url + "]: "
-					+ e.getMessage());
+			log.debug("Caught URISyntaxException while resolving url [" + url + "]: " + e.getMessage());
```
Comment (gvanbrakel) [inline tibco/src/main/java/nl/nn/adapterframework/extensions/tibco/GetTibcoQueues.java:577]: hier graag `, e` i.p.v. ` + e.getMessage()`

File: tibco/src/main/java/nl/nn/adapterframework/extensions/tibco/GetTibcoQueues.java [hunk 4]
```diff
@@ -601,16 +574,14 @@ private String getResolvedUrl(String url) {
...
 			return null;
 		}
 		InetAddress inetAddress = null;
 		try {
 			inetAddress = InetAddress.getByName(uri.getHost());
 		} catch (UnknownHostException e) {
-			log.debug("Caught UnknownHostException while resolving url [" + url + "]: "
-					+ e.getMessage());
+			log.debug("Caught UnknownHostException while resolving url [" + url + "]: " + e.getMessage());
```
Comment (gvanbrakel) [inline tibco/src/main/java/nl/nn/adapterframework/extensions/tibco/GetTibcoQueues.java:584]: hier graag `, e` i.p.v. ` + e.getMessage()`

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

**Review by gvanbrakel — APPROVED** [review]

**Pull request merged**